File: back/Relatorio.py
class Relatorio:
    def __init__(self, db):
        self.db = db

    def inserir_relatorio_inicial(self, descricao, id_avaliacao, caminho_arquivo):
        try:
            query = """
                INSERT INTO relatorio (Descricao, ID_Tipo, ID_Avaliacao, Caminho_Arquivo)
                VALUES (%s, %s, %s, %s)
            """
            values = (descricao, 1, id_avaliacao, caminho_arquivo)
            self.db.cursor.execute(query, values)
            self.db.conn.commit()
            return self.db.cursor.lastrowid
        except Exception as e:
            print(f"Erro ao inserir relatório: {e}")
            self.db.conn.rollback()
            raise

    def atualizar_relatorio_inicial(self, descricao, id_avaliacao, caminho_arquivo):
        try:
            query = """
                UPDATE relatorio
                SET Descricao = %s, Caminho_Arquivo = %s
                WHERE ID_Avaliacao = %s AND ID_Tipo = 1
            """
            values = (descricao, caminho_arquivo, id_avaliacao)
            self.db.cursor.execute(query, values)
            self.db.conn.commit()
        except Exception as e:
            print(f"Erro ao atualizar relatório: {e}")
            self.db.conn.rollback()
            raise

    def obter_relatorio_inicial(self, id_avaliacao):
        try:
            query = """
                SELECT Descricao, Caminho_Arquivo
                FROM relatorio
                WHERE ID_Avaliacao = %s AND ID_Tipo = 1
            """
            self.db.cursor.execute(query, (id_avaliacao,))
            result = self.db.cursor.fetchone()
            if result:
                return {"descricao": result[0], "caminhoArquivo": result[1]}
            return None
        except Exception as e:
            print(f"Erro ao obter relatório: {e}")
            raise
    
    # Métodos para a ata de reunião de abertura
    def inserir_ata_abertura(self, descricao, id_avaliacao):
        try:
            query = """
                INSERT INTO relatorio (Descricao, ID_Tipo, ID_Avaliacao, Caminho_Arquivo)
                VALUES (%s, %s, %s, %s)
            """
            values = (descricao, 2, id_avaliacao, 'não existe')
            self.db.cursor.execute(query, values)
            self.db.conn.commit()
            return self.db.cursor.lastrowid
        except Exception as e:
            print(f"Erro ao inserir ata de abertura: {e}")
            self.db.conn.rollback()
            raise

    def atualizar_ata_abertura(self, descricao, id_avaliacao):
        try:
            query = """
                UPDATE relatorio
                SET Descricao = %s, Caminho_Arquivo = %s
                WHERE ID_Avaliacao = %s AND ID_Tipo = 2
            """
            values = (descricao, 'não existe', id_avaliacao)
            self.db.cursor.execute(query, values)
            self.db.conn.commit()
        except Exception as e:
            print(f"Erro ao atualizar ata de abertura: {e}")
            self.db.conn.rollback()
            raise

    def obter_ata_abertura(self, id_avaliacao):
        try:
            query = """
                SELECT Descricao, Caminho_Arquivo
                FROM relatorio
                WHERE ID_Avaliacao = %s AND ID_Tipo = 2
            """
            self.db.cursor.execute(query, (id_avaliacao,))
            result = self.db.cursor.fetchone()
            if result:
                return {"descricao": result[0], "caminhoArquivo": result[1]}
            return None
        except Exception as e:
            print(f"Erro ao obter ata de abertura: {e}")
            raise

File: backjs/classes/Relatorio.js
class Relatorio {
    constructor(db) {
        this.db = db;
    }

    async inserirRelatorioInicial(descricao, idAvaliacao, caminhoArquivo) {
        const query = `
            INSERT INTO relatorio (Descricao, ID_Tipo, ID_Avaliacao, Caminho_Arquivo)
            VALUES (?, ?, ?, ?)
        `;
        const values = [descricao, 1, idAvaliacao, caminhoArquivo];
        try {
            await this.db.executeQuery(query, values);
            const lastId = await this.db.getLastInsertId(); // Função que retorna o último ID inserido
            return lastId;
        } catch (err) {
            console.error("Erro ao inserir relatório inicial:", err);
            throw err;
        }
    }

    async atualizarRelatorioInicial(descricao, idAvaliacao, caminhoArquivo) {
        const query = `
            UPDATE relatorio
            SET Descricao = ?, Caminho_Arquivo = ?
            WHERE ID_Avaliacao = ? AND ID_Tipo = 1
        `;
        const values = [descricao, caminhoArquivo, idAvaliacao];
        try {
            await this.db.executeQuery(query, values);
        } catch (err) {
            console.error("Erro ao atualizar relatório inicial:", err);
            throw err;
        }
    }

    async obterRelatorioInicial(idAvaliacao) {
        const query = `
            SELECT Descricao, Caminho_Arquivo
            FROM relatorio
            WHERE ID_Avaliacao = ? AND ID_Tipo = 1
        `;
        try {
            const result = await this.db.fetchOne(query, [idAvaliacao]);
            if (result) {
                return { descricao: result.Descricao, caminhoArquivo: result.Caminho_Arquivo };
            }
            return null;
        } catch (err) {
            console.error("Erro ao obter relatório inicial:", err);
            throw err;
        }
    }

    // Métodos para ata de reunião de abertura
    async inserirAtaAbertura(descricao, idAvaliacao) {
        const query = `
            INSERT INTO relatorio (Descricao, ID_Tipo, ID_Avaliacao, Caminho_Arquivo)
            VALUES (?, ?, ?, ?)
        `;
        const values = [descricao, 2, idAvaliacao, 'não existe'];
        try {
            await this.db.executeQuery(query, values);
            const lastId = await this.db.getLastInsertId(); // Função que retorna o último ID inserido
            return lastId;
        } catch (err) {
            console.error("Erro ao inserir ata de abertura:", err);
            throw err;
        }
    }

    async atualizarAtaAbertura(descricao, idAvaliacao) {
        const query = `
            UPDATE relatorio
            SET Descricao = ?, Caminho_Arquivo = ?
            WHERE ID_Avaliacao = ? AND ID_Tipo = 2
        `;
        const values = [descricao, 'não existe', idAvaliacao];
        try {
            await this.db.executeQuery(query, values);
        } catch (err) {
            console.error("Erro ao atualizar ata de abertura:", err);
            throw err;
        }
    }

    async obterAtaAbertura(idAvaliacao) {
        const query = `
            SELECT Descricao, Caminho_Arquivo
            FROM relatorio
            WHERE ID_Avaliacao = ? AND ID_Tipo = 2
        `;
        try {
            const result = await this.db.fetchOne(query, [idAvaliacao]);
            if (result) {
                return { descricao: result.Descricao, caminhoArquivo: result.Caminho_Arquivo };
            }
            return null;
        } catch (err) {
            console.error("Erro ao obter ata de abertura:", err);
            throw err;
        }
    }
}

module.exports = Relatorio;